contracts/src/token/erc721/extensions/consecutive.rs

```diff
@@ -0,0 +1,1079 @@
+
+            if emit_event {
+                evm::log(Approval { owner, approved: to, token_id });
+            }
+        }
+
+        self.erc721._token_approvals.setter(token_id).set(to);
+        Ok(())
+    }
+
+    /// Reverts if the `token_id` doesn't have a current owner (it hasn't been
+    /// minted, or it has been burned). Returns the owner.
+    ///
+    /// Overrides to ownership logic should be done to
+    /// [`Self::_owner_of_inner`].
+    ///
+    /// # Errors
+    ///
+    /// If token does not exist, then the error
+    /// [`Error::NonexistentToken`] is returned.
+    ///
+    /// # Arguments
+    ///
+    /// * `&self` - Read access to the contract's state.
+    /// * `token_id` - Token id as a number.
+    pub fn _require_owned(&self, token_id: U256) -> Result<Address, Error> {
+        let owner = self._owner_of_inner(token_id);
+        if owner.is_zero() {
+            return Err(erc721::Error::NonexistentToken(
+                ERC721NonexistentToken { token_id },
+            )
+            .into());
+        }
+        Ok(owner)
+    }
+}
+
+#[cfg(all(test, feature = "std"))]
+mod test {
+    use alloy_primitives::{address, uint, Address, U256};
+    use stylus_sdk::{msg, prelude::StorageType};
+
+    use crate::{
+        token::{
+            erc721,
+            erc721::{
+                extensions::consecutive::{
+                    ERC721ExceededMaxBatchMint, ERC721ForbiddenBatchMint,
+                    Erc721Consecutive, Error, MAX_BATCH_SIZE,
+                },
+                tests::random_token_id,
+                ERC721InvalidReceiver, ERC721NonexistentToken, Erc721, IErc721,
+            },
+        },
+        utils::structs::{
+            bitmap::BitMap,
+            checkpoints::{Trace160, U96},
+        },
+    };
+
+    // TODO#q: add support for complex contracts creation for motsu
+    impl Default for Erc721Consecutive {
+        fn default() -> Self {
+            #[derive(Default)]
+            struct StorageTypeFactory {
+                root: U256,
+            }
+            impl StorageTypeFactory {
+                fn create<T: StorageType>(&mut self) -> T {
+                    let instance = unsafe { T::new(self.root, 0) };
+                    self.root += U256::from(T::SLOT_BYTES);
+                    instance
+                }
+            }
+
+            let mut factory = StorageTypeFactory::default();
+            Erc721Consecutive {
+                erc721: Erc721 {
+                    _owners: factory.create(),
+                    _balances: factory.create(),
+                    _token_approvals: factory.create(),
+                    _operator_approvals: factory.create(),
+                },
+                _sequential_ownership: Trace160 {
+                    _checkpoints: factory.create(),
+                },
+                _sequentian_burn: BitMap { _data: factory.create() },
+                _initialized: factory.create(),
+            }
+        }
+    }
```

Comment:
btw, I removed this code, and added the `#[cfg_attr(all(test, feature = "std"), derive(motsu::DefaultStorageLayout))]` to `Erc721Consecutive` and all the tests pass.

Do you mind sharing the issues you were having? cc: @ggonzalez94

Reply:
Tough to say. I suspect smth else was wrong or I hasn't checked and added default implementation. But yeah. It works just because `StorageType::new` for `erc721` was recursively implemented with `sol_storage!` macro. And that was unforeseen by us before.